src/envoy/auth/jwt.cc

```diff
@@ -0,0 +1,176 @@
+#include "jwt.h"
+
+#include "common/common/base64.h"
+#include "openssl/evp.h"
+#include "openssl/rsa.h"
+#include "rapidjson/document.h"
+
+#include <algorithm>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace Envoy {
+namespace Http {
+namespace Auth {
+
+namespace Base64url {
+
+std::string decode(std::string input) {
+  std::replace(input.begin(), input.end(), '-', '+');
+  std::replace(input.begin(), input.end(), '_', '/');
+  switch (input.length() % 4) {
+    case 0:
+      break;
+    case 2:
+      input += "==";
+      break;
+    case 3:
+      input += "=";
+      break;
+    default:
+      /* invalid base64url input*/
+      assert(0);
+  }
+  return Base64::decode(input);
+}
+
+}  // namespace Base64url
+
+namespace Util {
+
+uint8_t* unsigned_c_str(const std::string& str) {
+  return reinterpret_cast<uint8_t*>(const_cast<char*>(str.c_str()));
```

Comment:
Why do you need to do a `const_cast` here? It is unsafe and I think the BoringSSL functions takes `const uint8_t*`s.